File: backend/model/paginated.go
package model

type PaginatedMeta struct {
	Total       int64 `json:"total"`
	LastPage    int32 `json:"lastPage"`
	CurrentPage int32 `json:"currentPage"`
	PerPage     int32 `json:"perPage"`
	Prev        int32 `json:"prev"`
	Next        int32 `json:"next"`
}

type Paginated[T any] struct {
	Data []T           `json:"data"`
	Meta PaginatedMeta `json:"meta"`
}

File: frontend/src/model/paginated.ts
/**
 * PaginatedDto is a generic DTO for paginated responses.
 */
export interface Paginated<T = any> {
  data: T[]
  meta: {
    total: number
    lastPage: number
    currentPage: number
    perPage: number
    prev: number | null
    next: number | null
  }
}
